gptme/message.py

```diff
@@ -126,39 +124,24 @@ def _content_files_list(
                 )
                 continue
 
-            if provider == "anthropic":
-                content.append(
-                    {
-                        "type": "image",
-                        "source": {
-                            "type": "base64",
-                            "media_type": media_type,
-                            "data": data,
-                        },
-                    }
-                )
-            elif provider == "openai":
-                # OpenAI format
-                content.append(
-                    {
-                        "type": "image_url",
-                        "image_url": {"url": f"data:{media_type};base64,{data}"},
-                    }
-                )
+            file_content = get_model().prepare_file(media_type, data)
+            if file_content:
+                content.append(file_content)
             else:
                 # Storage/wire format (keep files in `files` list)
                 # Do nothing to integrate files into the message content
                 pass
 
         return content
 
-    def to_dict(self, keys=None, provider: Provider | None = None) -> dict:
+    def to_dict(self, keys=None) -> dict:
         """Return a dict representation of the message, serializable to JSON."""
         content: str | list[dict[str, Any]]
-        if provider in ProvidersWithFiles:
+
+        if get_model().supports_file:
             # OpenAI/Anthropic format should include files in the content
             # Some otherwise OpenAI-compatible providers (groq, deepseek?) do not support this
-            content = self._content_files_list(provider)
+            content = self._content_files_list()
```

Comment:
Same as above, get rid of this from `message.py` and move it into provider logic after running msgs2dicts.